File: aoc6.hs
import qualified Data.Set as S
import System.Environment

main :: IO ()
main = do
    args <- getArgs
    let filename = if null args then "aoc6.in" else head args
    s <- readFile filename
    print $ head $ filter (\idx -> S.size (S.fromList (drop (idx - 4) $ take idx s)) == 4) [4..]
    print $ head $ filter (\idx -> S.size (S.fromList (drop (idx - 14) $ take idx s)) == 14) [14..]

File: aoc2022/aoc6.py
from aoc_util import get_data

data = get_data(6)
idx = 4
while len(set(data[idx - 4 : idx])) < 4:
    idx += 1
print(idx)

idx = 14
while len(set(data[idx - 14 : idx])) < 14:
    idx += 1
print(idx)
